### DISCUSSION TITLE
working in cmd.exe as a command (posh-windows-amd64.exe)

### DISCUSSION THREAD

**Discussion by IBIT-ZEE** [JanDeDobbeleer/oh-my-posh#1460] [Troubleshoot]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What would you like to see changed/added?

Hi,,,
I tested with CLink to get a CMD.EXE prompt with OMP 
but I run into a few problems... 

I'm trying to get OMP info with just cmd.exe,,, 

I've made a few tweaks with the folders and themes
and create a OMP-CMD that calls  "posh-windows-amd64.exe"
with the theme parameter ( I'm using "jandedobbeleer.omp.json" )

so when I want the omp info-line...  I just type omp (omp.cmd)
(my main goal for now is to get the github info... next... we'll see...)

it works fine but no git information...

what is the mechanism used to get the git information...
(in the posh-windows-amd64.exe executable)
?is via gh.exe... git.exe... or other ?

Thanks...

**Comment by JanDeDobbeleer** [comment]
It uses git.exe under the hood. Let me get this right, you're not following the guide but followed your own path? Because it's difficult to debug something non-standard. You do need the git segment in your config, so make sure that's there

**Reply by IBIT-ZEE** [reply]
Thanks for the feedback...

I see,,, I was trying to avoid analyzing the code,,, but...

(...)
g.gitCommand = "git" 
    if g.env.getRuntimeGOOS() == windowsPlatform || inWSL2SharedDrive(g.env)
        {  g.gitCommand = "git.exe" }
(...)

That could be a problem because some use git/gh.exe...
finding the executable for sure facilitates things for the standard user,
but can be a problem for other cases...

the problem ca arise also with other languages
(i'm thinking of python.exe in python2/3... or similar cases)

there is also the case you can have some versions of the same compiler
for testing purposes and you call them by an alias... or a script...

Anyway... your approach if laudable but if I can suggest an improvement...
a config file (ini/json/...) in the OMP folder that is read before the find function
where we can declare the executables/scripts for each supported feature:

something like:
git=c:\apl\git64\bin\git.exe
python=c:\apl\python3\python.exe
etc. etc.

By the way... my problem was solved as soon as I put the git in the path
(my path variable are usually clean for security reasons...)
if possible I rely on scripts/alias to call the programs...
my git.cmd makes a few verifications before invoke the git.exe...

So now in cmd.exe I can invoke the omp.cmd when I want
and get your excellent information prompt (mainly for git purposes...)

I'm considering put the this solution in github
as soon as I can spare a few hours

Thanks again, ZEE

**Reply by JanDeDobbeleer** [reply]
A few things, `git.exe` and `gh.exe` are not the same executable. They're completely different. The `python` segment has that logic built-in so there's no issue there either, it favours 3 over 2, but can handle them all. We also support almost every version management tool out there, so that's also not really an issue. Aliae are not supported as we can't use them, they're shell specific, not accessible from go. If you do avoid using the `PATH` entirely, I would argue oh-my-posh is not made for you. The use-case where you call oh-my-posh from a `.cmd` file is also not why it's been designed.
